bluesky_widgets/qt/search_results.py

```diff
@@ -69,7 +77,7 @@ def run(self):
             # requires some measurement. For now, the performance of this
             # implementation is acceptable. Most important, the application
             # does not lock up at all during data loading.
-            self._data_changed.emit(index, index, [])
+            self._signals.data_changed.emit(index, index, [])
```

Comment:
It might make sense to not perform self._data[index] = item and rather emit the item with the index and perform the assignment back in the main thread